homeassistant/components/aladdin_connect/__init__.py

```diff
@@ -35,4 +31,6 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
 
 async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Unload a config entry."""
-    return await hass.config_entries.async_unload_platforms(entry, PLATFORMS)
+    unload_ok = await hass.config_entries.async_unload_platforms(entry, PLATFORMS)
+    hass.data[DOMAIN].pop(entry.entry_id)
```

Comment:
We normally only pop from hass.data if the unload is successful.

https://github.com/home-assistant/core/blob/e8664ce1aee95bb71f941be9de83f8f2bb6a74de/script/scaffold/templates/config_flow/integration/__init__.py#L25-L30

Reply:
Shouldn't we have a test for "failed unload" to verify not popping the entry?  I don't see any other integrations testing for this situation.

Reply:
It's not clear what state we expect the integration to have in this case. Some of the unloading will have been done while some of it will have failed. The user will be prompted to restart Home Assistant.

We also want to avoid asserting integration details like hass.data.

I'd leave it for now.